engine/BUILD.bazel

```diff
@@ -0,0 +1,66 @@
+load("//:version.bzl", "PROJECT_VERSION")
+
+cc_library(
+    name = "enginelib",
+    hdrs = glob(["src/**/*.hpp"], exclude=["src/main*.hpp"]),
+    srcs = glob(
+        ["src/**/*.cpp"],
+        exclude=[
+            "src/main*.cpp",
+            "src/**/*_test.cpp",
+            "src/server_mock.cpp",
+        ]
+    ),
+    data = glob(["lua/**"]),
+    includes = ["src"],
+    additional_compiler_inputs = glob(["webui/**"]),
+    deps = [
+        "//engine/vendor/lrdb",
+        "//engine/vendor/linenoise",
```

Comment:
I know that is not possible to that cleanly, while also supporting cmake but consider consuming all vendor files (linenoise and so on) as bazel modules in the long run (you then just have cleanly documented all third-party dependencies that way as well)

Reply:
Agreed, but that would require more changes to all the CMake/Conan stuff, and I want to add experimental Bazel tooling that works with the sources as-is.